src/realm/object-store/c_api/socket_provider.cpp

```diff
@@ -29,22 +31,48 @@ struct CAPITimer : sync::SyncSocketProvider::Timer {
         realm_release(m_handler);
     }
 
-    /// Cancel the timer immediately.
+    // Cancel the timer immediately - the CAPI implementation will need to call the
+    // realm_sync_socket_timer_canceled function to notify the sync client that the
+    // timer has been canceled and must be called in the same execution thread as
+    // the timer complete.
     void cancel() override
     {
         m_timer_cancel(m_userdata, m_timer);
     }
 
 private:
+    // A pointer to the CAPI implementation's timer instance. This is provided by the
+    // CAPI implementation when the create_timer_func function is called.
     realm_sync_socket_timer_t m_timer = nullptr;
 
-    realm_userdata_t m_userdata = nullptr;
+    // A wrapped reference to the callback function to be called when the timer completes,
+    // is canceled or an error occurs. This is provided by the Sync Client
     realm_sync_socket_callback_t* m_handler = nullptr;
+
+    // These values were originally provided to the socket_provider instance by the CAPI
+    // implementation when it was created
+    realm_userdata_t m_userdata = nullptr;
```

Comment:
I don't think this is initialized via the ctor, it is also passed to  https://github.com/realm/realm-core/blob/f56d7700909d0a58aa674c4cc4159cebb1c47d66/src/realm/object-store/c_api/socket_provider.cpp#L29-L30

Reply:
This is an overlook, so it should be.